core/src/main/java/org/eclipse/hono/config/AbstractConfig.java

```diff
@@ -228,20 +228,34 @@ public final void setKeyStorePassword(final String keyStorePassword) {
      */
     public KeyCertOptions getKeyCertOptions() {
 
-        if (keyPath != null && certPath != null
-                && (keyFormat == PEM || (hasPemFileSuffix(keyPath) && hasPemFileSuffix(certPath)))) {
+        final FileFormat format;
+
+        if (keyPath != null && certPath != null) {
+            format = FileFormat.orDetect(keyFormat, keyPath).orElse(null);
+
+            final FileFormat certFormat = FileFormat.orDetect(keyFormat, certPath).orElse(null);
+            if (certFormat != FileFormat.PEM) {
```

Comment:
Is this about certFormat being not the same as format or is this about any of them not being PEM? At least, we are not checking format being PEM, do we?

Reply:
The logic should be, if "keyPath" is set, then "certPath" has to be set as well. And both have to be PEM. Either detected or specified. That is what I read out of the original logic.

Maybe I mixed something up though.

Reply:
It seems as if *format* could (in theory) be something else than `PEM` while *certFormat* is `PEM`, couldn't it?
I think the code works because it relies on the fact that `FileFormat` support PEM only (for single key files). An explicit check for `format == certFormat` would make our intention clearer here FMPOV.

Reply:
`certFormat` could become something different than PEM if "orDetect" gets called with `null` as input format (auto detection) and the auto detection does return something different than PEM. e.g. you configure auto detection and specify a PKCS12 file as `certPath`.

Reply:
keyFormat is `null` by default.
If I do `props.setKeyPath("myKey.jks")` and `props.setCertPath("myCert.pem"), then all checks in `props.getKeyCertOptions()` pass but we end up with an exception because there is no keyStorePath set, or am I mistaken?

Reply:
True. Indeed the format when using `keyPath` and `certPath` should be enforced to using PEM.

Reply:
I guess so, yes.